scripts/infra/cloud-instance.sh

```diff
@@ -291,11 +293,12 @@ install_rh_nvidia_drivers() {
     local cloud_type=$1
     "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" "sudo dnf install kernel-devel-\$(uname -r) kernel-headers-\$(uname -r) -y"
     "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf install https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm -y
-    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf config-manager --add-repo https://developer.download.nvidia.com/compute/cuda/repos/rhel9/x86_64/cuda-rhel9.repo
-    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf clean expire-cache
-    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf module install nvidia-driver:latest-dkms -y
-    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf install cuda-toolkit nvidia-gds libnccl-devel libcudnn8 nvidia-driver-NVML nvidia-driver-cuda-libs -y
-    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf update -y
+    "${BASH_SOURCE[0]}" "$cloud_type" ssh -n "$INSTANCE_NAME" sudo dnf install https://developer.download.nvidia.com/compute/cuda/12.4.0/local_installers/cuda-repo-rhel9-12-4-local-12.4.0_550.54.14-1.x86_64.rpm -y
```

Comment:
It might be good to leave a comment here with an explanation about why these specific versions are pinned, or what should be checked before someone tries to change it.

Reply:
Agree, both vllm and the PyTorch version we are targeting (2.3.0) need cuda 12.4 (aren't compatible with 12.5 yet)

Reply:
Comment added, thanks for the suggestion!